common/types.go

```diff
@@ -574,3 +574,20 @@ func (c *Credentials) Parse(ctx context.Context, credentials goja.Value) error {
 	}
 	return nil
 }
+
+// HTTPHeader is a single HTTP header.
+type HTTPHeader struct {
+	Name  string `json:"name"`
+	Value string `json:"value"`
+}
+
+// HTTPMessageSize are the sizes in bytes of the HTTP message header and body.
+type HTTPMessageSize struct {
+	Headers int64 `json:"headers"`
+	Body    int64 `json:"body"`
+}
+
+// Total returns the total size in bytes of the HTTP message.
+func (s HTTPMessageSize) Total() int64 {
+	return s.Headers + s.Body
+}
```

Comment:
This `types` file seems too generic, and probably all elements defined in it could be moved somewhere else. But focusing on these modifications, would it make sense to move these types to a separate file? Such as `http.go` maybe?

Reply:
Sure 👍

Reply:
See: 24ed386 and 62255c2